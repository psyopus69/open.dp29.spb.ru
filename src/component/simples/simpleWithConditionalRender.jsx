import React from "react";
import {PropTypes} from "prop-types";

const SimpleWithConditionalRender = (props) => {
  const {renderCondition} = props;
  return <React.Fragment>{renderCondition(-1)}{renderCondition(0)}{renderCondition(1)}
  </React.Fragment>;
};
SimpleWithConditionalRender.propTypes = {
  renderCondition: PropTypes.func.isRequired,
};

export {
  SimpleWithConditionalRender,
};
