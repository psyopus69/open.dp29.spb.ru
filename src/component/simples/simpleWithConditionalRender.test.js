import React from "react";
import renderer from "react-test-renderer";
import {SimpleWithConditionalRender} from "./simpleWithConditionalRender.jsx";

it(`panel with setTimerAction & clear it`, () => {
  const tree = renderer.create(<SimpleWithConditionalRender
    addCondition={jest.fn()}
    clearCondition={jest.fn()}
    renderCondition={jest.fn()}
  />);
  expect(tree).toMatchSnapshot();
});
