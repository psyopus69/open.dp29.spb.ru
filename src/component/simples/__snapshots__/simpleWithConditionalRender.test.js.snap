// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panel with setTimerAction & clear it 1`] = `null`;
